File: src/main/java/ir/reservs/reservs/ui/base/BaseContract.java
package ir.reservs.reservs.ui.base;

public class BaseContract {
    public interface Presenter<T> {
        void onAttach(T view);

        void onDetach();
    }

    public interface View {
        void onError(String msg);

        void onError(int resId);

    }
}

File: src/main/java/ir/reservs/reservs/ui/base/BaseContract.kt
package ir.reservs.reservs.ui.base;

class BaseContract {
    interface View {
        fun onError(msg: String)
        fun onError(resId: Int)
    }

    interface Presenter<T> {
        fun onAttach(view: T)
        fun onDetach()
    }
}